### PR TITLE
Add a build option to enable source level coverage analysis

### PR DESCRIPTION
**Pull request by zhaomaosu**

### CODE REVIEW

File: options_compile.cpp [hunk 1]
```diff
@@ -161,6 +161,7 @@ std::string EffectiveOptionsFilter::processOptions(const OpenCLArgList &args,
       // The below assert is usable for manual debugging only
       // default:
       // assert(false && "some unknown argument");
+    case OPT_COMPILE_coverage:
```
Comment (wenju-he) [inline options_compile.cpp:164]: is line-tables-only enough for coverage?

**Comment by haonanya** [comment]
Build failure should be fixed by spirv https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/2209.

File: opencl_clang_options.td [hunk 2]
```diff
@@ -14,6 +14,7 @@ include "llvm/Option/OptParser.td"
 def g_Flag : Flag<["-"], "g">, HelpText<"Generate source level debug information">;
 def gline_tables_only_Flag : Flag<["-"], "gline-tables-only">;
 def profiling : Flag<["-"], "profiling">, HelpText<"Allow profiling">;
+def coverage: Flag<["-"], "coverage">, HelpText<"Enable source level coverage analysis">;
```
Comment (wenju-he) [inline opencl_clang_options.td:17]:
```suggestion
def coverage: Flag<["-"], "code_coverage">, HelpText<"Enable source level coverage analysis">;
```
is code_coverage better?

File: options_compile.cpp (hunk 1 above)
Comment (zhaomaosu) [reply options_compile.cpp:164]: Yes, line-tables-only is enough. Code coverage doesn't need other information like type.

File: opencl_clang_options.td (hunk 2 above)
Comment (zhaomaosu) [reply opencl_clang_options.td:17]: clang/gcc both use '-coverage'.

**Review by wenju-he — APPROVED** [review]
LGTM

**Pull request merged**